[webkitpy] Apply timeouts to individual simctl commands (Follow-up)

JonWBedard opened the pull request:
#### 20ae11673d56cd78a3d5aec5fcfc6be4768e143f
<pre>
[webkitpy] Apply timeouts to individual simctl commands (Follow-up)
<a href="https://bugs.webkit.org/show_bug.cgi?id=302826">https://bugs.webkit.org/show_bug.cgi?id=302826</a>
<a href="https://rdar.apple.com/165085170">rdar://165085170</a>

Reviewed by Elliott Williams.

&apos;simctl terminate&apos; relies on simulator frames executing, if a simulator has completey hanged, we
will never get a simulator frame, so &apos;simctl terminate&apos; will never resolve.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDeviceManager._wait_until_device_is_usable): Support a response of &quot;None&quot; from
&quot;is_usable&quot;, which indicates a device will never become usable.
(SimulatedDevice.is_usable): Return &quot;None&quot; if &apos;simctl terminate&apos; times out, since this means
the simulated device has completely hanged.

Canonical link: <a href="https://commits.webkit.org/303354@main">https://commits.webkit.org/303354@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2623c38d5c4517ad268043eb803f52aafcad0851

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132132 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4625 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43154 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139647 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/84050 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/47563959-f69e-4348-9fc4-758bbc74f0b7) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134002 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4573 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4386 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/101005 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/84050 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/876087a1-8f96-480f-8ebe-1769ba29c7bc) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135078 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3245 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118348 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81796 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/63314f4e-f2fa-4a98-a882-0e81c69c8e88) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/131481 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3137 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/1046 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82866 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/111920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36471 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142293 "Built successfully") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/153/builds/4294 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37047 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/109380 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4375 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/3739 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109554 "Passed tests") | | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/164/builds/3262 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114623 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57540 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20539 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4348 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/33016 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4179 "Built successfully") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/4439 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4307 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

bls1999 commented on Tools/Scripts/webkitpy/xcode/simulated_device.py:
```diff
@@ -636,9 +640,18 @@ def is_usable(self, force_update=False):
+        try:
+            with Timeout(seconds=30, patch=False):
+                exit_code |= self.executive.run_command([SimulatedDeviceManager.xcrun, 'simctl', 'terminate', self.udid, service], return_exit_code=True)
+        except Timeout.Exception:
```
Why not just raise straight from here?

bls1999 reviewed (approved):
[unofficial] r+ with comment

emw-apple commented on Tools/Scripts/webkitpy/xcode/simulated_device.py:
```diff
@@ -363,10 +363,14 @@ def _wait_until_device_in_state(device, state, deadline):
+        state = device.platform_device.is_usable(force_update=True)
+        while not state:
+            if state is None:
+                raise RuntimeError(u'{} will never become healthy'.format(device))
```
nit: We should never be running this on any python version below 3.9, so you can modernize the string if you want.
```suggestion
                raise RuntimeError(f'{device} will never become healthy')
```

JonWBedard committed:
[webkitpy] Apply timeouts to individual simctl commands (Follow-up)
https://bugs.webkit.org/show_bug.cgi?id=302826
rdar://165085170

Reviewed by Elliott Williams.

'simctl terminate' relies on simulator frames executing, if a simulator has completey hanged, we
will never get a simulator frame, so 'simctl terminate' will never resolve.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDeviceManager._wait_until_device_is_usable): Support a response of "None" from
"is_usable", which indicates a device will never become usable.
(SimulatedDevice.is_usable): Return "None" if 'simctl terminate' times out, since this means
the simulated device has completely hanged.

Canonical link: https://commits.webkit.org/303354@main
